src/con_duct/__main__.py

```diff
@@ -208,15 +210,18 @@ def max(self: Sample, other: Sample) -> Sample:
         return output
 
     def for_json(self) -> dict[str, Any]:
-        d = {str(pid): asdict(stats) for pid, stats in self.stats.items()}
-        d["totals"] = {
-            "pmem": self.total_pmem,
-            "pcpu": self.total_pcpu,
-            "rss_kb": self.total_rss,
-            "vsz_kb": self.total_vsz,
+        d = {
+            "timestamp": self.timestamp,
+            "num_samples": self.averages.num_samples,
+            "processes": {str(pid): asdict(stats) for pid, stats in self.stats.items()},
+            "totals": {
+                "pmem": self.total_pmem,
+                "pcpu": self.total_pcpu,
+                "rss_kb": self.total_rss,
+                "vsz_kb": self.total_vsz,
+            },
+            "averages": asdict(self.averages) if self.averages.num_samples >= 1 else {},
```

Comment:
yes, this looks cleaner.

I guess unrelated -- what is "totals"? aren't they "max" through the time so we could see how much needed?  Where is total wallclock run time (in seconds)? is there a way to get total cpu run time as well?

Reply:
"max through time" is not recorded in `usage.json`. The usage file contains nothing about the full run, it is only (aggregated) samples. 

The "totals" in this case are total of all processes. ie, if the inner command forks and is running multiple processes simultaneously, the `totals` will report on the command as a whole, rather than per process. 


The total wallclock run time is recorded in the `info.json`. Since we are recording wall clock time and average cpu utilization, do we need cpu time? If so, lets file a separate issue.

{
  "command": "sleep 5",
  "system": {
    "uid": "austin",
    "memory_total": 33336778752,
    "cpu_total": 20
  },
  "env": {},
  "gpu": null,
  "duct_version": "0.0.1",
  "execution_summary": {
    "exit_code": 0,
    "command": "sleep 5",
    "logs_prefix": "ZZZZ_",
    "wall_clock_time": "5.003 sec",
    "peak_rss": "1920 KiB",
    "average_rss": "1920.000 KiB",
    "peak_vsz": "221584 KiB",
    "average_vsz": "221584.000 KiB",
    "peak_pmem": "0.0%",
    "average_pmem": "0.000%",
    "peak_pcpu": "0.0%",
    "average_pcpu": "0.000%",
    "num_samples": 4,
    "num_reports": 3
  }
}